addons/mail/models/mail_thread.py

```diff
@@ -1388,11 +1388,6 @@ def message_process(self, model, message, custom_values=None,
         # convert it to utf-8 for transport between the mailgate script and here.
         if isinstance(message, xmlrpclib.Binary):
             message = bytes(message.data)
-        # message_from_string parses from a *native string*, except apparently
-        # sometimes message is ISO-8859-1 binary data or some shit and the
-        # straightforward version (pycompat.to_native) won't work right ->
-        # always encode message to bytes then use the relevant method
-        # depending on ~python version
         if isinstance(message, str):
             message = message.encode('utf-8')
         extract = getattr(email, 'message_from_bytes', email.message_from_string)
```

Comment:
There's no need for the dispatch between message_from_bytes and message_from_string anymore, it's a p2/p3 thing (message_from_string is the original but misleading method, message_from_bytes was added in 3.2).